detector/test_helper.go

```diff
@@ -66,14 +67,29 @@ func TestDetectByCreatorName(creatorMethod string, src string, stateJSON string,
 	if preprocess := detector.MethodByName("PreProcess"); preprocess.IsValid() {
 		preprocess.Call([]reflect.Value{})
 	}
-	for _, template := range schema {
-		for _, resource := range template.Find("resource", reflect.Indirect(detector).FieldByName("Target").String()) {
-			detect := detector.MethodByName("Detect")
-			detect.Call([]reflect.Value{
-				reflect.ValueOf(resource),
-				reflect.ValueOf(issues),
-			})
+	switch reflect.Indirect(detector).FieldByName("TargetType").String() {
+	case "resource":
+		for _, template := range schema {
+			for _, resource := range template.FindResources(reflect.Indirect(detector).FieldByName("Target").String()) {
+				detect := detector.MethodByName("Detect")
+				detect.Call([]reflect.Value{
+					reflect.ValueOf(resource),
+					reflect.ValueOf(issues),
+				})
+			}
 		}
+	case "module":
```

Comment:
__[gometalinter]__
* [goconst] 2 other occurrence(s) of "module" found in: detector.go:162:7 detector.go:231:7

<!-- Comment by [SideCI](https://sideci.com) -->